system/View/Table.php

```diff
@@ -295,14 +295,17 @@ public function generate($tableData = null)
 
         // Is there a table heading to display?
         if (! empty($this->heading)) {
+            preg_match('/(<)(td|th)/i', $this->template['heading_cell_start'], $valueHead);
+            $strHeading = count($valueHead) === 3 ? $valueHead[1] . $valueHead[2] : null;
+
             $out .= $this->template['thead_open'] . $this->newline . $this->template['heading_row_start'] . $this->newline;
 
             foreach ($this->heading as $heading) {
                 $temp = $this->template['heading_cell_start'];
 
                 foreach ($heading as $key => $val) {
                     if ($key !== 'data') {
-                        $temp = str_replace('<th', '<th ' . $key . '="' . $val . '"', $temp);
+                        $temp = $strHeading ? str_replace($strHeading, $strHeading . ' ' . $key . '="' . $val . '"', $temp) : $temp;
```

Comment:
This ternary condition can be moved up to the `if` above to check if `$strHeading !== null`